cmd/tpm-manager/tpm-manager.go

```diff
@@ -0,0 +1,506 @@
+		&cache.ListWatch{
+			ListFunc: func(options api.ListOptions) (runtime.Object, error) {
+				return manager.tpmhandler.PolicyClient.List(&options)
+			},
+			WatchFunc: func(options api.ListOptions) (watch.Interface, error) {
+				return manager.tpmhandler.PolicyClient.Watch(&options)
+			},
+		},
+		&runtime.Unstructured{},
+		controller.NoResyncPeriodFunc(),
+		framework.ResourceEventHandlerFuncs{
+			AddFunc:    policyAddFn,
+			UpdateFunc: policyUpdateFn,
+			DeleteFunc: policyDeleteFn,
+		},
+	)
+	fmt.Printf("Starting\n")
+	getPolicy()
+	go reverify()
+	go nodeController.Run(wait.NeverStop)
+	go configController.Run(wait.NeverStop)
+	go policyController.Run(wait.NeverStop)
+	select {}
+	fmt.Printf("Returned\n")
+}
+
+func main() {
+	var tpmhandler tpm.TPMHandler
+	var config restclient.Config
+
+	//	config, err := clientcmd.BuildConfigFromFlags(manager.Master, manager.Kubeconfig)
+	err := client.SetKubernetesDefaults(&config)
+	if err != nil {
+		fmt.Errorf("Unable to create client configuration: %v", err)
+	}
+	config.Host = "http://localhost:8080"
+	client, err := client.New(&config)
+	if err != nil {
+		fmt.Printf("Unable to create client: %v", err)
+		return
+	}
+	tpmhandler.Setup(&config)
+
+	configmap, err := client.ConfigMaps("default").Get("tpm-manager.coreos.com")
+	if err == nil && configmap != nil {
+		updateConfig(configmap)
+	}
+
+	addFlags(pflag.CommandLine)
+	leaderelection.BindFlags(&manager.leaderelect, pflag.CommandLine)
+	flag.InitFlags()
+
+	manager.client = client
+	manager.tpmhandler = tpmhandler
+	manager.recurringChan = make(chan int)
+
+	if !manager.leaderelect.LeaderElect {
+		run(nil)
+	}
+
+	id, err := os.Hostname()
+	if err != nil {
+		fmt.Printf("Unable to obtain hostname: %v", err)
+		return
+	}
+
+	eventBroadcaster := record.NewBroadcaster()
+	eventBroadcaster.StartLogging(glog.Infof)
+	eventBroadcaster.StartRecordingToSink(client.Events(""))
+	recorder := eventBroadcaster.NewRecorder(api.EventSource{Component: "tpm-manager"})
+
+	leaderelection.RunOrDie(leaderelection.LeaderElectionConfig{
+		EndpointsMeta: api.ObjectMeta{
+			Namespace: "",
+			Name:      "tpm-manager",
+		},
+		Client:        client,
+		Identity:      id,
+		EventRecorder: recorder,
+		LeaseDuration: manager.leaderelect.LeaseDuration.Duration,
+		RenewDeadline: manager.leaderelect.RenewDeadline.Duration,
+		RetryPeriod:   manager.leaderelect.RetryPeriod.Duration,
+		Callbacks: leaderelection.LeaderCallbacks{
+			OnStartedLeading: run,
+			OnStoppedLeading: func() {
+				glog.Fatalf("leaderelection lost")
+			},
+		},
+	})
+}
+
+func nodeAddFn(obj interface{}) {
+	fmt.Printf("New node\n")
+	node, ok := obj.(*api.Node)
+	if !ok {
+		return
+	}
+	fmt.Printf("Verifying\n")
+	verifyAndUpdate(node)
+}
+
+func nodeUpdateFn(oldObj, newObj interface{}) {
+}
+
+func nodeDeleteFn(obj interface{}) {
+}
+
+func configAddFn(obj interface{}) {
+	fmt.Printf("New config\n")
+	configmap, ok := obj.(*api.ConfigMap)
+	if !ok {
+		return
```

Comment:
nit: Really the informer is misconfigured if we get the wrong type here. But might be worth putting a high-verbosity logline just in case (protects against accidentally using wrong Addfn for wrong informer.